### PR TITLE
Change labels structure

### PR DESCRIPTION
**Pull request by BANOnotIT**
Notable changes:
- `bug` -> `T - Defect`
- `docs` -> `P - Docs`
- `help wanted` -> `good first issue`
- `refactor` -> `T - Task`
- `proposal` -> `P - Proposal`
- `investigate` -> `need info`

Labels to be deleted:
- `tests`
- `types`
- `question`
- `invalid`
- `design`
- `examples`

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+blank_issues_enabled: true
+contact_links:
+  - name: Reatom Community Support
+    url: https://github.com/artalar/reatom/discussions
+    about: Please ask and answer questions here.
+#  - name: GitHub Security Bug Bounty
+#    url: https://bounty.github.com/
+#    about: Please report security vulnerabilities here.
```
Comment (BANOnotIT) [inline .github/ISSUE_TEMPLATE/config.yaml]: @artalar I need your info for this

**Comment by BANOnotIT** [comment]
@artalar so how do we proceed with this? Do you need any more info or any feedback?

**Comment by BANOnotIT** [comment]
https://github.com/marketplace/actions/github-labeler might be a good idea to have this app installed

**Comment by artalar** [comment]
@BANOnotIT I'm finally ready to dive into this :) Will you update any conflicts and other things if needed?

**Comment by BANOnotIT** [comment]
Uhm... I would consider this thing as wip since I found a solution that can manage tags automatically

**Comment by BANOnotIT** [comment]
Could you review the tag system first? I mean the structure and check it there are any problems I might missed

**Comment by artalar** [comment]
@BANOnotIT my thoughts

- `bug` - for unexpected hurt runtime behavior of a library, it should be obvious.
- `problem` - for types issue, CI&CD problems, docs styles issues and so on. I think we should separate this kind of issues and `bug`s to prevent misunderstood about the quality of the product.
- `docs` is fine.
- `good first issue` is fine.
- `refactor` / `task` - don't get it, isn't is `proposal`?
- `proposal` is fine, but it could be `task` too.
- `need info` is fine.

What about `CI` also?

**Comment by BANOnotIT** [comment]
@artalar 
- I propose using general Type `Defect` instead of `bug` and `problem` because we can define what exactly is behaving abnormally using Parts: package, docs or types. That's the main idea behind deviding Type and Parts
- `refactor` becomes `T - Task`. Tasks should be treated as a piece of work to be done: move files around, setup tests, change CI parts and other things.
- `T - Proposal` should not be Task. Proposal should be a discussion issue that implementing Tasks could be lined to.
- `CI` is Part.

Also in rust repo they use [labels without spaces](https://forge.rust-lang.org/release/triage-procedure.html) in format X-dead-beef with leading uppercased type letter and other part of the label kebab-cased to be easily handled by bots and people in filters. I guess that's a great UX thing. What do you think?

**Comment by artalar** [comment]
@BANOnotIT Hi! I don't want to create many labels. My experience in a regular job shows me that for actual work, `task` and `bug` is sufficient.

However, as we are a public project, we should have a couple of additional labels.

The `help wanted` label is needed to indicate that the issue could be addressed by an external contributor.

The `problem` / `defect` (up to you) label is needed for bugs that are not relevant to user functionality, such as documentation, CI, types or so on. This distinction is important for our marketing perspective.

**Comment by BANOnotIT** [comment]
Well I think unless we do some automation dependent on labels we don't really need a Scheme.

Can you elaborate how bug and separate problem/defect affects marketing?

Also, I wanted to rename "help wanted" because "good first issue" had a separate view in GitHub and we can employ it.

**Comment by artalar** [comment]
@BANOnotIT "Bug" is a common label that a new user investigates when trying to evaluate a library. If we have extra `bug`s for docs, CSS, CI, and so on, it will present the library in the wrong light, literally in the wrong way.

`good first issue` is ok, you are right.

**Comment by artalar** [comment]
@BANOnotIT one more thing. All proposals and feature requests should starts from https://github.com/artalar/reatom/discussions

**Comment by artalar** [comment]
@BANOnotIT ping

**Comment by BANOnotIT** [comment]
I added [RFCs category](https://github.com/artalar/reatom/discussions/categories/rfcs) in discussions but by default people will be creating Ideas discussions.

I also propose following actions for current labels:
https://github.com/artalar/reatom/labels/CI%26CD, https://github.com/artalar/reatom/labels/design, https://github.com/artalar/reatom/labels/docs, https://github.com/artalar/reatom/labels/tests just to be deleted

https://github.com/artalar/reatom/labels/proposal, https://github.com/artalar/reatom/labels/feature%20request, https://github.com/artalar/reatom/labels/investigate converted into Ideas discussions

New labels to be added are described in new CONTRIBUTING.md

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ If you want to contribute to improving the library, use the following instructio
 
 ## Creating an issue
 
-If you found a bug or want to make an improvement in the library please check whether the same issue already exists in the [list of issues](https://github.com/artalar/reatom/issues). If you don't find the issue there, [create a new one](https://github.com/artalar/reatom/issues/new) including a description of the problem.
+If you found a bug or want to make an improvement in the library please check whether the same issue already exists in the [list of issues](https://github.com/artalar/reatom/issues). If you don't find the issue there, [create a new one](https://github.com/artalar/reatom/issues/new) including a description of the problem. Note [labels used](#labels-used-in-repository).
```
Comment (artalar) [inline CONTRIBUTING.md:13]:
```suggestion
If you found a bug or want to make an improvement in the library please check whether the same issue already exists in the [list of issues](https://github.com/artalar/reatom/issues). If you don't find the issue there, [create a new one](https://github.com/artalar/reatom/issues/new) including a description of the problem. Note [labels used](#labels-used-in-repository).

If you have a question or want to propose a feature or an improvement, please create [a discussion](https://github.com/artalar/reatom/discussions).
```

**Review by artalar — APPROVED** [review]

**Pull request merged**